src/modules/account/internal/account-param-validation.ts

```diff
@@ -0,0 +1,46 @@
+import zxcvbn from 'zxcvbn';
+import {
+  CreateAccountParams,
+  ValidationError,
+  ValidationFailure,
+} from '../types';
+import emailRegex from '../constants';
+
+export default class AccountParamValidation {
+  public static validate(params: CreateAccountParams): void {
+    const failures: ValidationFailure[] = [];
+    this.isUsernameValid(params.username, failures);
+    this.isPasswordValid(params.password, failures);
+    if (failures.length) {
+      throw new ValidationError(
+        'Account cannot be created, please check the params validity.',
+        failures,
+      );
+    }
+  }
+
+  public static isUsernameValid(email: string, failures: ValidationFailure[]): void {
+    const emailValid = emailRegex.test(String(email).toLowerCase());
+    if (!emailValid) {
+      failures.push({
+        field: 'username',
+        message: 'Please specify valid email.',
+      });
+    }
+  }
+
+  public static isPasswordValid(
+    password: string,
+    failures: ValidationFailure[],
+  ): void {
+    const passwordStrength = zxcvbn(password);
+    if (passwordStrength.score < 3) {
```

Comment:
It is not clear what `3` stands for.

1. Can you add some comments in the code and reference to other literature what `3` means?
2. Can you also add comment why `3`?